tools/cpp/runfiles/runfiles_src.h

```diff
@@ -160,21 +180,48 @@ class Runfiles {
     return envvars_;
   }
 
+  // Returns a new Runfiles instance that by default uses the provided source
+  // repository as a default for all calls to Rlocation.
+  //
+  // The current instance remains valid.
+  Runfiles* WithSourceRepository(const std::string& source_repository) const {
+    std::map<std::string, std::string> runfiles_map(runfiles_map_);
+    std::string directory(directory_);
+    std::map<std::pair<std::string, std::string>, std::string>
+        repo_mapping(repo_mapping_);
+    std::vector<std::pair<std::string, std::string> > envvars(envvars_);
+    auto* new_runfiles =
+        new Runfiles(std::move(runfiles_map), std::move(directory),
+                     std::move(repo_mapping), std::move(envvars),
+                     std::string(source_repository));
+    return new_runfiles;
+  }
+
  private:
-  Runfiles(const std::map<std::string, std::string>&& runfiles_map,
-           const std::string&& directory,
-           const std::vector<std::pair<std::string, std::string> >&& envvars)
+  Runfiles(std::map<std::string, std::string>&& runfiles_map,
```

Comment:
just make these take the bare type (`std::map<...>` instead of `std::map<...>&&`)

Reply:
Fixed.